runtime/lua/vim/lsp/_snippet.lua

```diff
@@ -173,32 +173,33 @@ P.seq = function(...)
   end
 end
 
-local Node = {}
-
-Node.Type = {
-  SNIPPET = 0,
```

Comment:
👍  say no to uppercase (spicy zero-likes tweet: https://twitter.com/justinmk/status/1669461598786711552 )

Reply:
I still don't understand what's so offensive about ASCII values 65-90. Does your font have particularly bad glyphs for this range or something?

Reply:
I see it as a cargo-cult. If that's the case, then it's obvious why it's bothersome :)

Reply:
> I obviously don't agree that it reduces to being offended by a character range.

On that twitter thread you wrote:

> Do prefer kOhShitHereComesAKonstant? 
>
> sure, the k is less noisy than uppercase + underscores. 
>
> why does this single property of a variable matter so much compared to all other possible dimensions ?

I can only read this as a distaste for consecutive ASCII 65-90,95, whilst also suggesting that extra unrelated `k`'s are ok 🤔.

I've explained this before....

Being able to easily determine whether something is state or not, statically, **is** a very important property in all programming domains. I hope that's something we can agree on. Using uppercase for this has simply become a popular convention. If there are other, better, ways to do it then I'm sure the cult is open to suggestions.

Reply:
> I can only read this as a distaste for consecutive ASCII 65-90,95, whilst also suggesting that extra unrelated `k`'s are ok 🤔.

No, it's a distaste for cargo cults. 

> Being able to easily determine whether something is state or not, statically, is a very important property in all programming domains.

In this case the symbols are part of a table. An enum is similar. Both are statically analyzable.

Reply:
> No, it's a distaste for cargo cults.

Up until now, I've not seen any evidence presented that this is one. I've only seen your opinion that you think const-ness isn't important when it comes to readability, which I deeply disagree with. I refactored enough code at this point to know this. Further repeating "cargo cult" (on its own) makes this increasingly frustrating.

> Both are statically analyzable.

We have no tools other than `@enum` for that, and because of that we don't uppercase enums nor do we have a convention to indicate whether a symbol is a table or not. Inlay hints can do that for us, and even clangd now does a very good job with semantic tokens at differentiating const-ness without the need for a naming convention.

In this case it's unlikely `SNIPPET` is going to be used as a symbol by itself and is likely to be prefixed with `Node.type.`, so it's unlikely that convention is beneficial here. And as you suggest, this could be an `@enum`.

Reply:
> even clangd now does a very good job with semantic tokens at differentiating const-ness without the need for a naming convention.

We're in violent agreement? 😅

Reply:
Lua 5.1, unlike C (or even Javascript) does not have language level attributes for const-ness, and LusLS currently has no `@const` annotation. So.... our only choice to provide such a distinction to code readers is via naming convention. The same applies to python, bash, tcl, etc.